frontendFile/src/main/java/com/virtuslab/gitmachete/frontend/file/grammar/Machete.flex

```diff
@@ -0,0 +1,58 @@
+// Copyright 2000-2020 JetBrains s.r.o. and other contributors. Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE file.
+package com.virtuslab.gitmachete.frontend.file.grammar;
+
+import com.intellij.lexer.FlexLexer;
+import com.intellij.psi.tree.IElementType;
+import com.intellij.psi.TokenType;
+
+%%
+
+%class MacheteLexer
+%implements FlexLexer
+%unicode
+%function advance
+%type IElementType
+%eof{  return;
+%eof}
+
+
+// End Of Line
+EOL=[\r\n]
+// For sake of distinguishablility WHITESPACE and INDENTATION are two separated entries
+// even if they are the same for now
+WHITESPACE=[\ \t]
+INDENTATION=[\ \t]
+// Probably all characters allowed in branch name
+// (https://mirrors.edge.kernel.org/pub/software/scm/git/docs/git-check-ref-format.html)
+NAME_WITHOUT_SLASH=[\w\-.!@#$%&()+={}\];'\",.<>|`]
+// In addition, names can contain / (slash)
+NAME_WITH_SLASH=[\w\-.!@#$%&()+={}\];'\",.<>|`\/]
```

Comment:
Hmm what's `\w` here?

Reply:
Maybe drop a comment to make it instantly obvious

Reply:
`\w` is standard regex abbreviation of `[a-zA-Z0-9_]`